system-addon/lib/TopStoriesFeed.jsm

```diff
@@ -307,11 +403,21 @@ this.TopStoriesFeed = class TopStoriesFeed {
           this.spocs = this.spocs.filter(s => s.url !== action.data.url);
         }
         break;
+      case at.TELEMETRY_IMPRESSION_STATS:
```

Comment:
This action not only handles the impressions, but also takes care of clicks&blocks. I guess we have to only pick up the impression ones here.

Reply:
Ah good catch! I checked for action.tiles, thinking that's only the case for views, but clicks etc. have tiles too, just an array of length 1 then.

Reply:
Yep, the easy fix would be check if the length of `action.data.tiles` === TOPSTORIES_SIZE

Reply:
Had to actually filter the event "types" because users could dismiss all but two stories, then tiles would be an array of length 2...